### PR TITLE
Task 7

### PR DESCRIPTION
**Pull request by Moluram**

### CODE REVIEW

File: task7/CheckNumber.java [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+ */
+public class CheckNumber {
+  /**
+   * Takes line from user input and send it to various tests
+   * @param args - arguments from command line
+   */
+  public static void main(String[] args) {
+    String line = new Scanner(System.in).nextLine().replace(","," ");
+    CheckLine checkLine = new CheckLine();
+    if (checkLine.forContainMoreThan5Words(line)) {
```
Comment (AlexanderSoroka) [inline task7/CheckNumber.java]: What if you have 1k rules? will you add more and more checks?

File: task7/checker/Checker.java [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+
+/**
+ * Interfaces for classes which wants to check string line
+ */
+interface Checker {
+  /**
+   * Main function for checkers
+   * Return result of check
+   * @param line - checked line
+   * @return boolean
```
Comment (AlexanderSoroka) [inline task7/checker/Checker.java]: Return value MUST be described

File: task7/checker/ContainMoreThan5Words.java [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+   * Checks whether the given string contain more than five words
+   * Return answer
+   * @param line - line for check
+   * @return boolean
+   */
+  public boolean check(String line) {
+    String[] words = line.split(" ");
+    int counter = 0;
+    for (String word: words) {
+        if (word.length()!=0) {
```
Comment (AlexanderSoroka) [inline task7/checker/ContainMoreThan5Words.java]: Code style violation

File: task7/checker/ContainMoreThan5Words.java [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+          counter++;
+      }
+    }
+    return counter > 5;
```
Comment (AlexanderSoroka) [inline task7/checker/ContainMoreThan5Words.java]: It doesn't generic and flexible, does it?

File: task7/checker/ContainOnlyDigits.java [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+  /**
+   * Checks whether the given string consists only of digits
+   * Return answer
+   * @param line - line for testing
+   * @return boolean
+   */
+  public boolean check(String line) {
+    String pattern = "[^0-9]";
+    Pattern p = Pattern.compile(pattern);
+    Matcher m = p.matcher(line.replace(" ", ""));
```
Comment (AlexanderSoroka) [inline task7/checker/ContainOnlyDigits.java]: Wrong implementation - it also will be positive for line with digits and spaces

File: task7/checker/DoNotContainDigits.java [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+class DoNotContainDigits implements Checker {
+  /**
+   * Checks whether the given line contain digits
+   * Prints the answer
+   * @param line - checked line
+   */
+  public void check(String line) {
+    String pattern = "[0-9]";
+    Pattern p = Pattern.compile(pattern);
+    Matcher m = p.matcher(line.replace(" ",""));
```
Comment (AlexanderSoroka) [inline task7/checker/DoNotContainDigits.java]: Why do you need to drop off all spaces?

File: task7/checker/ContainWordPalindrome.java [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+ * @version 1.0
+ */
+class ContainWordPalindrome implements Checker {
+  /**
+   * Checks whether the given string contain palindrome word
+   * Prints the answer
+   * @param line - line for check
+   */
+  public void check(String line) {
+    for (String word: line.split(" ")) {
```
Comment (AlexanderSoroka) [inline task7/checker/ContainWordPalindrome.java]: Specification violation

File: task7/CheckLine.java [hunk 8]
```diff
@@ -0,0 +1,67 @@
...
+  private List<Checker> listOfCheckers = new ArrayList<>();
+
+  public CheckLine() {
+    createListOfCheckers();
+    FileInputStream fis;
+    try {
+      fis = new FileInputStream("src/com/moluram/task7/outputStrings.properties");
+      property.load(fis);
+    } catch (IOException e) {
+      System.err.println("ОШИБКА: Файл свойств отсуствует!");
```
Comment (AlexanderSoroka) [inline task7/CheckLine.java]: Don't use Russian anywhere

File: task7/checker/ContainMoreThanNWords.java [hunk 9]
```diff
@@ -0,0 +1,35 @@
...
+ * Class serve to answer the question :
+ * Is this line contain more than N words?
+ * @author Moluram
+ * @version 1.0
+ */
+public class ContainMoreThanNWords implements Checker {
+  /**
+   * Number of words
+   */
+  private int N;
```
Comment (AlexanderSoroka) [inline task7/checker/ContainMoreThanNWords.java]: codestyle violation

File: task7/checker/ContainMoreThanNWords.java [hunk 10]
```diff
@@ -0,0 +1,35 @@
...
+
+  public ContainMoreThanNWords(int N){
```
Comment (AlexanderSoroka) [inline task7/checker/ContainMoreThanNWords.java]: codestyle violation

File: task7/checker/ContainMoreThanNWords.java [hunk 11]
```diff
@@ -0,0 +1,35 @@
...
+    this.N = N;
+  }
+
+  /**
+   * Checks whether the given string contain more than five words
+   * @param line - line for check
+   * @return boolean - answer of the check
+   */
+  public boolean check(String line) {
+    String[] words = line.split(" ");
```
Comment (AlexanderSoroka) [inline task7/checker/ContainMoreThanNWords.java]: Specification violation

Comment (Moluram) [inline task7/checker/ContainMoreThanNWords.java]: What's the problem there?

Comment (AlexanderSoroka) [inline task7/checker/ContainMoreThanNWords.java]: You split the line only by "," and space and make in very non-obvious and error-prone way. Actually Your rules (every of it) have to suppose that words are splitted only by spaces that is wrong. It's much better to parametrize rules (checker in your terms) by some "split rule"

File: task7/checker/ContainWordOfLength4ThatNextToMaxLengthWord.java [hunk 12]
```diff
@@ -0,0 +1,59 @@
...
+  }
+
+  /**
+   * Finds answer  for the question:
+   * Is that array contain word of length 4 that nex to max length word
+   * @param words - array of words
+   * @param maxLength - max length of the words from "words" array
+   * @return boolean - answer to the question
+   */
+  private boolean isContainWordOfLength4ThatNextToMaxLengthWord(String[] words, int maxLength) {
```
Comment (AlexanderSoroka) [inline task7/checker/ContainWordOfLength4ThatNextToMaxLengthWord.java]: First time I see magic number not just in code but in Method name! It's a very dangerous - if you need to change method behavior you have to make changes in at least 3 places - method name, comment and some constant. Such changes can make code very messy

**Pull request merged**